lib/matplotlib/table.py

```diff
@@ -211,12 +243,17 @@ def __init__(self, ax, loc=None, bbox=None, **kwargs):
         self.set_clip_on(False)
 
         self._cachedRenderer = None
+        self._cellType = 'default'
 
     def add_cell(self, row, col, *args, **kwargs):
         """ Add a cell to the table. """
         xy = (0, 0)
 
-        cell = Cell(xy, *args, **kwargs)
+        if self.AVAILABLECELLTYPES[self._cellType] == 1:
+            cell = SciCell(xy, *args, **kwargs)
+        else:
+            cell = Cell(xy, *args, **kwargs)
```

Comment:
and if you are using a property, you should use it here instead of touching the internals.